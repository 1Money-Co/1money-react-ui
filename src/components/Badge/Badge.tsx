import { memo } from 'react';
import { Badge as PrimeBadge } from 'primereact/badge';
import classnames from '@/utils/classnames';
/* import types */
import type { FC, PropsWithChildren } from 'react';
import type { BadgeProps } from './interface';

export const Badge: FC<PropsWithChildren<BadgeProps>> = props => {
  const { children, className, severity, prefixCls = 'badge', ...rest } = props;
  const classes = classnames(prefixCls);

  return (
    <PrimeBadge
      {...rest}
      className={classes(void 0, [classes(severity || 'neutral'), className].join(' '))}
    />
  );
};

export default memo(Badge);
