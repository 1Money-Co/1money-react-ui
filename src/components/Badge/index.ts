import Badge from './Badge';

export { Badge } from './Badge';

export default Badge;

export * from './interface';
