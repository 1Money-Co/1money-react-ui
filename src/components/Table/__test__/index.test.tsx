import 'jsdom-global/register';
import * as React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { Table } from '../index';
import { Badge } from '../../Badge';
import { Icons } from '../../Icons';

const originalConsoleError = console.error;
console.error = (message, ...optionalParams) => {
  if (
    message.includes('Could not parse CSS stylesheet') ||
    message.includes('findDOMNode is deprecated and will be removed')
  ) {
      return;
  }
  originalConsoleError(message, ...optionalParams);
};

describe('Table', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <Table
      columns={
        [
          { field: 'nickname', header: 'Wallet nickname' },
          { field: 'asset', header: 'Asset' },
          { field: 'network', header: 'Network' },
          { field: 'address', header: 'Address' },
          { field: 'updatedAt', header: 'Update Time' },
          { field: 'status', header: 'Status', body: (data) => <Badge severity={data.status.severity} value={data.status.value} /> },
          {
            field: 'action', header: 'Action', body: (data) => <span style={{ display: 'inline-flex', alignItems: 'center', gap: '12px' }}>
              <Icons
                name='editFile'
                size={20}
              />
              <Icons
                name='remove'
                size={20}
              />
            </span>
          },
        ]
      }
      value={[
        {
          dataKey: 1,
          nickname: 'My income address 1',
          asset: 'USD1',
          network: 'Ethereum',
          address: '0x7c4b****25s2',
          updatedAt: '25 Jan 2025, 12:30',
          status: {
            value: 'Completed',
            severity: 'success',
          },
          action: 1,
        },
        {
          dataKey: 2,
          nickname: 'My income address 2',
          asset: 'USDC',
          network: 'Polygon',
          address: '0x7c4b****25s2',
          updatedAt: '25 Jan 2025, 12:30',
          status: {
            value: 'In Progress',
            severity: 'warning',
          },
          action: 2,
        }
      ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
