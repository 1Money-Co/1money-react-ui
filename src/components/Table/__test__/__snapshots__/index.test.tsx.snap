// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="om-react-ui-table    p-datatable p-component p-datatable-responsive-scroll"
        data-pc-name="datatable"
        data-pc-section="root"
        data-scrollselectors=".p-datatable-wrapper"
        data-showgridlines="false"
        pr_id_1=""
      >
        <div
          class="p-datatable-wrapper"
          data-pc-section="wrapper"
          style="overflow: auto;"
        >
          <table
            class="p-datatable-table"
            data-pc-section="table"
            role="table"
          >
            <thead
              class="p-datatable-thead"
              data-pc-name="datatable"
              data-pc-section="thead"
              role="rowgroup"
            >
              <tr
                data-pc-section="headerrow"
                role="row"
              >
                <th
                  data-p-filter-column="false"
                  data-p-frozen-column="false"
                  data-p-reorderable-column="false"
                  data-p-resizable-column="false"
                  data-p-sortable-column="false"
                  data-pc-name="datatable"
                  data-pc-section="headercell"
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                    data-pc-section="headercontent"
                  >
                    <span
                      class="p-column-title"
                      data-pc-section="headertitle"
                    >
                      Wallet nickname
                    </span>
                  </div>
                </th>
                <th
                  data-p-filter-column="false"
                  data-p-frozen-column="false"
                  data-p-reorderable-column="false"
                  data-p-resizable-column="false"
                  data-p-sortable-column="false"
                  data-pc-name="datatable"
                  data-pc-section="headercell"
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                    data-pc-section="headercontent"
                  >
                    <span
                      class="p-column-title"
                      data-pc-section="headertitle"
                    >
                      Asset
                    </span>
                  </div>
                </th>
                <th
                  data-p-filter-column="false"
                  data-p-frozen-column="false"
                  data-p-reorderable-column="false"
                  data-p-resizable-column="false"
                  data-p-sortable-column="false"
                  data-pc-name="datatable"
                  data-pc-section="headercell"
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                    data-pc-section="headercontent"
                  >
                    <span
                      class="p-column-title"
                      data-pc-section="headertitle"
                    >
                      Network
                    </span>
                  </div>
                </th>
                <th
                  data-p-filter-column="false"
                  data-p-frozen-column="false"
                  data-p-reorderable-column="false"
                  data-p-resizable-column="false"
                  data-p-sortable-column="false"
                  data-pc-name="datatable"
                  data-pc-section="headercell"
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                    data-pc-section="headercontent"
                  >
                    <span
                      class="p-column-title"
                      data-pc-section="headertitle"
                    >
                      Address
                    </span>
                  </div>
                </th>
                <th
                  data-p-filter-column="false"
                  data-p-frozen-column="false"
                  data-p-reorderable-column="false"
                  data-p-resizable-column="false"
                  data-p-sortable-column="false"
                  data-pc-name="datatable"
                  data-pc-section="headercell"
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                    data-pc-section="headercontent"
                  >
                    <span
                      class="p-column-title"
                      data-pc-section="headertitle"
                    >
                      Update Time
                    </span>
                  </div>
                </th>
                <th
                  data-p-filter-column="false"
                  data-p-frozen-column="false"
                  data-p-reorderable-column="false"
                  data-p-resizable-column="false"
                  data-p-sortable-column="false"
                  data-pc-name="datatable"
                  data-pc-section="headercell"
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                    data-pc-section="headercontent"
                  >
                    <span
                      class="p-column-title"
                      data-pc-section="headertitle"
                    >
                      Status
                    </span>
                  </div>
                </th>
                <th
                  data-p-filter-column="false"
                  data-p-frozen-column="false"
                  data-p-reorderable-column="false"
                  data-p-resizable-column="false"
                  data-p-sortable-column="false"
                  data-pc-name="datatable"
                  data-pc-section="headercell"
                  role="columnheader"
                >
                  <div
                    class="p-column-header-content"
                    data-pc-section="headercontent"
                  >
                    <span
                      class="p-column-title"
                      data-pc-section="headertitle"
                    >
                      Action
                    </span>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="p-datatable-tbody"
              data-pc-section="tbody"
              role=" rowgroup"
            >
              <tr
                data-p-highlight="false"
                data-p-highlight-contextmenu="false"
                data-pc-section="bodyrow"
                role="row"
              >
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  My income address 1
                </td>
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  USD1
                </td>
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  Ethereum
                </td>
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  0x7c4b****25s2
                </td>
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  25 Jan 2025, 12:30
                </td>
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  <span
                    class="om-react-ui-badge om-react-ui-badge-success  p-badge p-component"
                    data-pc-name="badge"
                    data-pc-section="root"
                  >
                    Completed
                  </span>
                </td>
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  <span
                    style="display: inline-flex; align-items: center; gap: 12px;"
                  >
                    <i
                      class="om-react-ui-icons-wrapper"
                      style="color: rgb(19, 19, 19); width: 20px; height: 20px;"
                    >
                      <svg
                        class="om-react-ui-icons"
                        fill="none"
                        height="20"
                        stroke="none"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M14 2H7C5.89543 2 5 2.89543 5 4V6M14 2L21 9M14 2V8C14 8.55228 14.4477 9 15 9H21M21 9V20C21 21.1046 20.1046 22 19 22H17.5"
                          stroke-width="2"
                        />
                        <path
                          d="M13.8681 21.8681L10.9872 21.548L2.37974 12.0345L4.96376 9.45045L13.5418 18.9314L13.8681 21.8681Z"
                          stroke-width="2"
                        />
                      </svg>
                    </i>
                    <i
                      class="om-react-ui-icons-wrapper"
                      style="color: rgb(19, 19, 19); width: 20px; height: 20px;"
                    >
                      <svg
                        class="om-react-ui-icons"
                        fill="none"
                        height="20"
                        stroke="none"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M21.5001 4.5H16.8751V3.375C16.8751 2.67881 16.5985 2.01113 16.1063 1.51884C15.614 1.02656 14.9463 0.75 14.2501 0.75H9.7501C9.05391 0.75 8.38623 1.02656 7.89395 1.51884C7.40167 2.01113 7.1251 2.67881 7.1251 3.375V4.5H2.5V6.75H4.1251V19.5C4.1251 19.9973 4.32265 20.4742 4.67428 20.8258C5.02591 21.1775 5.50282 21.375 6.0001 21.375H18.0001C18.4974 21.375 18.9743 21.1775 19.3259 20.8258C19.6776 20.4742 19.8751 19.9973 19.8751 19.5V6.75H21.5001V4.5ZM9.3751 3.375C9.3751 3.27554 9.41461 3.18016 9.48494 3.10984C9.55527 3.03951 9.65065 3 9.7501 3H14.2501C14.3496 3 14.4449 3.03951 14.5153 3.10984C14.5856 3.18016 14.6251 3.27554 14.6251 3.375V4.5H9.3751V3.375ZM17.6251 19.125H6.3751V6.75H17.6251V19.125ZM10.8751 8.94057V17.1906H8.6251V8.94057H10.8751ZM15.3751 8.94057V17.1906H13.1251V8.94057H15.3751Z"
                        />
                      </svg>
                    </i>
                  </span>
                </td>
              </tr>
              <tr
                class="p-row-odd"
                data-p-highlight="false"
                data-p-highlight-contextmenu="false"
                data-pc-section="bodyrow"
                role="row"
              >
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  My income address 2
                </td>
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  USDC
                </td>
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  Polygon
                </td>
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  0x7c4b****25s2
                </td>
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  25 Jan 2025, 12:30
                </td>
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  <span
                    class="om-react-ui-badge om-react-ui-badge-warning  p-badge p-component"
                    data-pc-name="badge"
                    data-pc-section="root"
                  >
                    In Progress
                  </span>
                </td>
                <td
                  data-p-cell-editing="false"
                  data-p-editable-column="true"
                  data-p-frozen-column="false"
                  data-p-selectable-cell="false"
                  data-p-selection-column="false"
                  data-pc-name="datatable"
                  data-pc-section="bodycell"
                  role="cell"
                >
                  <span
                    style="display: inline-flex; align-items: center; gap: 12px;"
                  >
                    <i
                      class="om-react-ui-icons-wrapper"
                      style="color: rgb(19, 19, 19); width: 20px; height: 20px;"
                    >
                      <svg
                        class="om-react-ui-icons"
                        fill="none"
                        height="20"
                        stroke="none"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M14 2H7C5.89543 2 5 2.89543 5 4V6M14 2L21 9M14 2V8C14 8.55228 14.4477 9 15 9H21M21 9V20C21 21.1046 20.1046 22 19 22H17.5"
                          stroke-width="2"
                        />
                        <path
                          d="M13.8681 21.8681L10.9872 21.548L2.37974 12.0345L4.96376 9.45045L13.5418 18.9314L13.8681 21.8681Z"
                          stroke-width="2"
                        />
                      </svg>
                    </i>
                    <i
                      class="om-react-ui-icons-wrapper"
                      style="color: rgb(19, 19, 19); width: 20px; height: 20px;"
                    >
                      <svg
                        class="om-react-ui-icons"
                        fill="none"
                        height="20"
                        stroke="none"
                        viewBox="0 0 24 24"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M21.5001 4.5H16.8751V3.375C16.8751 2.67881 16.5985 2.01113 16.1063 1.51884C15.614 1.02656 14.9463 0.75 14.2501 0.75H9.7501C9.05391 0.75 8.38623 1.02656 7.89395 1.51884C7.40167 2.01113 7.1251 2.67881 7.1251 3.375V4.5H2.5V6.75H4.1251V19.5C4.1251 19.9973 4.32265 20.4742 4.67428 20.8258C5.02591 21.1775 5.50282 21.375 6.0001 21.375H18.0001C18.4974 21.375 18.9743 21.1775 19.3259 20.8258C19.6776 20.4742 19.8751 19.9973 19.8751 19.5V6.75H21.5001V4.5ZM9.3751 3.375C9.3751 3.27554 9.41461 3.18016 9.48494 3.10984C9.55527 3.03951 9.65065 3 9.7501 3H14.2501C14.3496 3 14.4449 3.03951 14.5153 3.10984C14.5856 3.18016 14.6251 3.27554 14.6251 3.375V4.5H9.3751V3.375ZM17.6251 19.125H6.3751V6.75H17.6251V19.125ZM10.8751 8.94057V17.1906H8.6251V8.94057H10.8751ZM15.3751 8.94057V17.1906H13.1251V8.94057H15.3751Z"
                        />
                      </svg>
                    </i>
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="om-react-ui-table    p-datatable p-component p-datatable-responsive-scroll"
      data-pc-name="datatable"
      data-pc-section="root"
      data-scrollselectors=".p-datatable-wrapper"
      data-showgridlines="false"
      pr_id_1=""
    >
      <div
        class="p-datatable-wrapper"
        data-pc-section="wrapper"
        style="overflow: auto;"
      >
        <table
          class="p-datatable-table"
          data-pc-section="table"
          role="table"
        >
          <thead
            class="p-datatable-thead"
            data-pc-name="datatable"
            data-pc-section="thead"
            role="rowgroup"
          >
            <tr
              data-pc-section="headerrow"
              role="row"
            >
              <th
                data-p-filter-column="false"
                data-p-frozen-column="false"
                data-p-reorderable-column="false"
                data-p-resizable-column="false"
                data-p-sortable-column="false"
                data-pc-name="datatable"
                data-pc-section="headercell"
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                  data-pc-section="headercontent"
                >
                  <span
                    class="p-column-title"
                    data-pc-section="headertitle"
                  >
                    Wallet nickname
                  </span>
                </div>
              </th>
              <th
                data-p-filter-column="false"
                data-p-frozen-column="false"
                data-p-reorderable-column="false"
                data-p-resizable-column="false"
                data-p-sortable-column="false"
                data-pc-name="datatable"
                data-pc-section="headercell"
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                  data-pc-section="headercontent"
                >
                  <span
                    class="p-column-title"
                    data-pc-section="headertitle"
                  >
                    Asset
                  </span>
                </div>
              </th>
              <th
                data-p-filter-column="false"
                data-p-frozen-column="false"
                data-p-reorderable-column="false"
                data-p-resizable-column="false"
                data-p-sortable-column="false"
                data-pc-name="datatable"
                data-pc-section="headercell"
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                  data-pc-section="headercontent"
                >
                  <span
                    class="p-column-title"
                    data-pc-section="headertitle"
                  >
                    Network
                  </span>
                </div>
              </th>
              <th
                data-p-filter-column="false"
                data-p-frozen-column="false"
                data-p-reorderable-column="false"
                data-p-resizable-column="false"
                data-p-sortable-column="false"
                data-pc-name="datatable"
                data-pc-section="headercell"
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                  data-pc-section="headercontent"
                >
                  <span
                    class="p-column-title"
                    data-pc-section="headertitle"
                  >
                    Address
                  </span>
                </div>
              </th>
              <th
                data-p-filter-column="false"
                data-p-frozen-column="false"
                data-p-reorderable-column="false"
                data-p-resizable-column="false"
                data-p-sortable-column="false"
                data-pc-name="datatable"
                data-pc-section="headercell"
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                  data-pc-section="headercontent"
                >
                  <span
                    class="p-column-title"
                    data-pc-section="headertitle"
                  >
                    Update Time
                  </span>
                </div>
              </th>
              <th
                data-p-filter-column="false"
                data-p-frozen-column="false"
                data-p-reorderable-column="false"
                data-p-resizable-column="false"
                data-p-sortable-column="false"
                data-pc-name="datatable"
                data-pc-section="headercell"
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                  data-pc-section="headercontent"
                >
                  <span
                    class="p-column-title"
                    data-pc-section="headertitle"
                  >
                    Status
                  </span>
                </div>
              </th>
              <th
                data-p-filter-column="false"
                data-p-frozen-column="false"
                data-p-reorderable-column="false"
                data-p-resizable-column="false"
                data-p-sortable-column="false"
                data-pc-name="datatable"
                data-pc-section="headercell"
                role="columnheader"
              >
                <div
                  class="p-column-header-content"
                  data-pc-section="headercontent"
                >
                  <span
                    class="p-column-title"
                    data-pc-section="headertitle"
                  >
                    Action
                  </span>
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="p-datatable-tbody"
            data-pc-section="tbody"
            role=" rowgroup"
          >
            <tr
              data-p-highlight="false"
              data-p-highlight-contextmenu="false"
              data-pc-section="bodyrow"
              role="row"
            >
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                My income address 1
              </td>
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                USD1
              </td>
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                Ethereum
              </td>
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                0x7c4b****25s2
              </td>
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                25 Jan 2025, 12:30
              </td>
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                <span
                  class="om-react-ui-badge om-react-ui-badge-success  p-badge p-component"
                  data-pc-name="badge"
                  data-pc-section="root"
                >
                  Completed
                </span>
              </td>
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                <span
                  style="display: inline-flex; align-items: center; gap: 12px;"
                >
                  <i
                    class="om-react-ui-icons-wrapper"
                    style="color: rgb(19, 19, 19); width: 20px; height: 20px;"
                  >
                    <svg
                      class="om-react-ui-icons"
                      fill="none"
                      height="20"
                      stroke="none"
                      viewBox="0 0 24 24"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M14 2H7C5.89543 2 5 2.89543 5 4V6M14 2L21 9M14 2V8C14 8.55228 14.4477 9 15 9H21M21 9V20C21 21.1046 20.1046 22 19 22H17.5"
                        stroke-width="2"
                      />
                      <path
                        d="M13.8681 21.8681L10.9872 21.548L2.37974 12.0345L4.96376 9.45045L13.5418 18.9314L13.8681 21.8681Z"
                        stroke-width="2"
                      />
                    </svg>
                  </i>
                  <i
                    class="om-react-ui-icons-wrapper"
                    style="color: rgb(19, 19, 19); width: 20px; height: 20px;"
                  >
                    <svg
                      class="om-react-ui-icons"
                      fill="none"
                      height="20"
                      stroke="none"
                      viewBox="0 0 24 24"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M21.5001 4.5H16.8751V3.375C16.8751 2.67881 16.5985 2.01113 16.1063 1.51884C15.614 1.02656 14.9463 0.75 14.2501 0.75H9.7501C9.05391 0.75 8.38623 1.02656 7.89395 1.51884C7.40167 2.01113 7.1251 2.67881 7.1251 3.375V4.5H2.5V6.75H4.1251V19.5C4.1251 19.9973 4.32265 20.4742 4.67428 20.8258C5.02591 21.1775 5.50282 21.375 6.0001 21.375H18.0001C18.4974 21.375 18.9743 21.1775 19.3259 20.8258C19.6776 20.4742 19.8751 19.9973 19.8751 19.5V6.75H21.5001V4.5ZM9.3751 3.375C9.3751 3.27554 9.41461 3.18016 9.48494 3.10984C9.55527 3.03951 9.65065 3 9.7501 3H14.2501C14.3496 3 14.4449 3.03951 14.5153 3.10984C14.5856 3.18016 14.6251 3.27554 14.6251 3.375V4.5H9.3751V3.375ZM17.6251 19.125H6.3751V6.75H17.6251V19.125ZM10.8751 8.94057V17.1906H8.6251V8.94057H10.8751ZM15.3751 8.94057V17.1906H13.1251V8.94057H15.3751Z"
                      />
                    </svg>
                  </i>
                </span>
              </td>
            </tr>
            <tr
              class="p-row-odd"
              data-p-highlight="false"
              data-p-highlight-contextmenu="false"
              data-pc-section="bodyrow"
              role="row"
            >
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                My income address 2
              </td>
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                USDC
              </td>
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                Polygon
              </td>
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                0x7c4b****25s2
              </td>
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                25 Jan 2025, 12:30
              </td>
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                <span
                  class="om-react-ui-badge om-react-ui-badge-warning  p-badge p-component"
                  data-pc-name="badge"
                  data-pc-section="root"
                >
                  In Progress
                </span>
              </td>
              <td
                data-p-cell-editing="false"
                data-p-editable-column="true"
                data-p-frozen-column="false"
                data-p-selectable-cell="false"
                data-p-selection-column="false"
                data-pc-name="datatable"
                data-pc-section="bodycell"
                role="cell"
              >
                <span
                  style="display: inline-flex; align-items: center; gap: 12px;"
                >
                  <i
                    class="om-react-ui-icons-wrapper"
                    style="color: rgb(19, 19, 19); width: 20px; height: 20px;"
                  >
                    <svg
                      class="om-react-ui-icons"
                      fill="none"
                      height="20"
                      stroke="none"
                      viewBox="0 0 24 24"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M14 2H7C5.89543 2 5 2.89543 5 4V6M14 2L21 9M14 2V8C14 8.55228 14.4477 9 15 9H21M21 9V20C21 21.1046 20.1046 22 19 22H17.5"
                        stroke-width="2"
                      />
                      <path
                        d="M13.8681 21.8681L10.9872 21.548L2.37974 12.0345L4.96376 9.45045L13.5418 18.9314L13.8681 21.8681Z"
                        stroke-width="2"
                      />
                    </svg>
                  </i>
                  <i
                    class="om-react-ui-icons-wrapper"
                    style="color: rgb(19, 19, 19); width: 20px; height: 20px;"
                  >
                    <svg
                      class="om-react-ui-icons"
                      fill="none"
                      height="20"
                      stroke="none"
                      viewBox="0 0 24 24"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M21.5001 4.5H16.8751V3.375C16.8751 2.67881 16.5985 2.01113 16.1063 1.51884C15.614 1.02656 14.9463 0.75 14.2501 0.75H9.7501C9.05391 0.75 8.38623 1.02656 7.89395 1.51884C7.40167 2.01113 7.1251 2.67881 7.1251 3.375V4.5H2.5V6.75H4.1251V19.5C4.1251 19.9973 4.32265 20.4742 4.67428 20.8258C5.02591 21.1775 5.50282 21.375 6.0001 21.375H18.0001C18.4974 21.375 18.9743 21.1775 19.3259 20.8258C19.6776 20.4742 19.8751 19.9973 19.8751 19.5V6.75H21.5001V4.5ZM9.3751 3.375C9.3751 3.27554 9.41461 3.18016 9.48494 3.10984C9.55527 3.03951 9.65065 3 9.7501 3H14.2501C14.3496 3 14.4449 3.03951 14.5153 3.10984C14.5856 3.18016 14.6251 3.27554 14.6251 3.375V4.5H9.3751V3.375ZM17.6251 19.125H6.3751V6.75H17.6251V19.125ZM10.8751 8.94057V17.1906H8.6251V8.94057H10.8751ZM15.3751 8.94057V17.1906H13.1251V8.94057H15.3751Z"
                      />
                    </svg>
                  </i>
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
